### DISCUSSION TITLE
How to use KEDA for custom objects managed by operators

### DISCUSSION THREAD

**Discussion by Jeffwan** [kedacore/keda#1832] [General] [answered]
Hi 

Seems KEDA has some scaling objects like deployment, job. My use case is to give KEAD scaler capacity to some other teams,  the challenge part is we don't have control on the resource type. other teams may write their own crds and operators. 

Currently, their operator can reconcile a `ScaledObject` for their application since they know all the details. But this solution couples two team's work together. 

I am checking if there's an easier way to attach this KEDA trait to deployments managed by CRDs?

**Comment by tomkerkhove** [comment, marked as answer]
So if I get it correctly, your team is in charge of hosting KEDA and the other app teams to scale their apps?

I think this works perfectly as you can install KEDA to scale anything in any namespace, and then the others can just craft their ScaledObjects. Can you elaborate a bit more on what your concern or issue is please?

**Reply by Jeffwan** [reply]
Hi @tomkerkhove  

> So if I get it correctly, your team is in charge of hosting KEDA and the other app teams to scale their apps?

Yes, you are right. We not only host KEDA but also want to provide application construction ability to app team. 

We actually like to use OAM to construct applications and KEDA is one of the most important trait. The goal is we can easily wrap the work from apps team and attach trait to their custom resources.  The challenge is their custom resource and operator is kind of blackbox to us.  

For example, they create a `DB` CRD and operator will create 2 statefulsets for this `DB` object.  Technically, I think I should build a `Component` (a new Workload type) for this `DB`, next, since I won't know the underneath kubernetes resources name/kind so that I can not attach `keda trait` to it. (KEDA needs detail information in ScaleObject).  That's the challenge part. 

I think the question will be similar like "If we don't want to couple KEDA objects reconciliation logic in operators , is there a pluggable way to  leverage KEDA"?

**Reply by zroubalik** [reply]
I am not sure that I understand your question correctly, but KEDA could target the `DB` CRD (if it implements /scale subresource) and the scaling could be done via this CR (as you said a blackbox), you don't have to know about the statefulsets or any other resources. Is that what you are asking for? Or you can create a separate operator, that will be watching and inspecting the newly created objects and will create ScaledObject based on the findings from these objects.

**Reply by Jeffwan** [reply]
>  Is that what you are asking for? 

Yes, that's what I asked.  I think it makes sense. I will try to ask app team to expose `scale` subresource and scale corresponding objects in their controller and we can just attach `scaledobjects` to top level CR. Thanks for the reply.
